### ISSUE TITLE
Create utility Node type guard checks class

### ISSUE THREAD

**Issue by dsherret** [dsherret/ts-simple-ast#22]
This should be done using code generation. It's easy to forget to add methods for these.

1. Code generate a mixin that contains all the isX() methods.
2. Apply the mixin to Node.
3. Write a script that ensures all the mixins return true for these methods.

Edit: I'm having second thoughts about this because it would spam the auto-complete. I'm thinking it might be better to put this on a utils class.

--------------

```typescript
import Ast, {TypeGuards} from "ts-simple-ast";

// ...

if (TypeGuards.isEnumDeclaration(node)) {
    // node is EnumDeclaration in here
}

if (TypeGuards.isBodiedNode(node)) {
    // node is Node & BodiedNode in here
}
```

This class should be completely code generated so that no unit tests are required (code generation is low risk for mistakes) and so that it doesn't need to be maintained (because maintaining it would be a pain!)

**Comment by dsherret** [comment]
Make sure to search the code for this issue to find places that should be updated once implemented.

**Comment by frederikschubert** [comment]
Maybe https://github.com/ajafff/tsutils could be used here?

**Comment by dsherret** [comment]
@frederikschubert that could be used in conjunction with this library on the wrapped node's `compilerNode` properties.

I managed to implement this by code generating this file based on having it look at the structure of the existing code:

https://github.com/dsherret/ts-simple-ast/blob/ce0c36e80798d0c689a99d67b7f4baa2311bc86b/src/utils/TypeGuards.ts

...so the file will automatically be updated based on changes to the library.

Anyway, this is implemented in 0.62.0.
